Makefile

```diff
@@ -1,72 +1,53 @@
-DOWNSTREAMTRANSFORMS = svc_client.sysl svc_error_types.sysl svc_types.sysl
-GRAMMAR=$(wildcard .tmp/sysl-go/codegen/grammars/go.gen.g)
-START=goFile
-
-
-# Always execute these with just `make`
-.PHONY: setup clean gen
-sysl: clean setup gen downstream format tmp
-
-# try to clone, then try to fetch and pull
-setup:
-	# Syncing server-lib to $(SERVERLIB)
-	git clone https://github.com/anz-bank/sysl-go/ $(SERVERLIB)/sysl-go || true  # Don't fail
-	cd  $(SERVERLIB)/sysl-go || true
-	mkdir -p $(TMP)/sysl-go/
-	mkdir -p ${outdir}/${app}
-	# Copying server-lib to $(TMP)
-	cp -rf $(SERVERLIB)/sysl-go $(TMP)/
-	$(foreach path, $(dependencies), $(shell mkdir -p ${outdir}/$(path)))
-    $(foreach path, $(app), $(shell mkdir -p ${outdir}/$(path)))
-
-# Generate files with internal git service
-gen:
-	$(foreach file, $(TRANSFORMS), $(shell sysl codegen --basepath=$(basepath)/${outdir}/ --transform $(TRANSLOCATION)/$(file) --grammar ${GRAMMAR} --start ${START} --outdir=${outdir}/${app} --app-name ${app} $(input)))
-
-downstream:
-	$(foreach file, $(DOWNSTREAMTRANSFORMS), $(foreach downstream, $(dependencies), $(shell sysl codegen --basepath=$(basepath)/${outdir}/ --transform $(TRANSLOCATION)/$(file) --grammar ${GRAMMAR} --start ${START} --outdir=${outdir}/${downstream} --app-name ${downstream} $(input))))
-
-format:
-	gofmt -s -w ${outdir}/${app}/*
-	goimports -w ${outdir}/${app}/*
-	$(foreach path, $(dependencies), $(shell gofmt -s -w ${outdir}/${path}/*))
-	$(foreach path, $(dependencies), $(shell goimports -w ${outdir}/${path}/*))
-	./replace.sh
-
-
-# Remove the tmp directory after
-tmp:
-	rm -rf $(TMP)
-
-# Remove the generated files
-clean:
-	rm -rf $(outdir)
\ No newline at end of file
+SYSL  =  ./syslw
+ARRAI  =  ./arraiw
+CODEGEN_ROOT = internal/gen
+SPECS_ROOT = specs/frontend/petdemo
+
+.PHONY: all
+gen: backend-gen frontend-gen
+
+# frontend code
+.PHONY: frontend-gen
+.PHONY: frontend
+frontend-gen: petdemo-gen
+
+.PHONY: Petdemo
+PETDEMO = Petdemo
+
+frontend = rest-service
+.PHONY: petdemo-gen
+petdemo-gen: $(PETDEMO) frontend
+	$(run-codegen)
+
+# backend code
+.PHONY: backend-gen
+.PHONY: backend
+backend-gen: petstore-gen flickr-gen
+
+.PHONY: Flickr
+FLICKR = Flickr
+
+.PHONY: flickr-gen
+flickr-gen: $(FLICKR) backend
+	$(run-codegen)
+
+.PHONY: Petstore
+PETSTORE = Petstore
+
+.PHONY: petstore-gen
+petstore-gen: $(PETSTORE) backend
+	$(run-backgen)
+
+SYSL_TEMPLATE_ROOT = github.com/anz-bank/sysl-template
+backend = rest-service
+
+define run-codegen
+	rm -rf $(CODEGEN_ROOT)/$(shell echo $< | tr A-Z a-z)
+	rm -rf $(SPECS_ROOT)/sysl.json
+	mkdir -p $(CODEGEN_ROOT)/$(shell echo $< | tr A-Z a-z)
+	$(SYSL) pb --mode json $(SPECS_ROOT)/petdemo.sysl >> $(SPECS_ROOT)/sysl.json
+	$(ARRAI) run service_stub.arrai \
+		$(SYSL_TEMPLATE_ROOT)/$(CODEGEN_ROOT) $(SPECS_ROOT)/sysl.json \
+		$< $($(word 2,$^)) | tar xf - -C $(CODEGEN_ROOT)/$(shell echo $< | tr A-Z a-z)
+	goimports -w $(CODEGEN_ROOT)/$(shell echo $< | tr A-Z a-z)
+endef
```

Comment:
I noticed that you removed the `gofmt` and `goimports` directives. Do Arrai Transforms produce formatted Go code with the right import statements?

Reply:
yes